static/usr/lib/udev/rules.d/66-snapd-autoimport.rules

```diff
@@ -1,3 +1,2 @@
-# probe for assertions, must run before udisks2
-ACTION=="add", SUBSYSTEM=="block", KERNEL!="loop*", KERNEL!="ram*" \
-    RUN+="/usr/bin/unshare -m /usr/bin/snap auto-import --mount=/dev/%k"
+ACTION=="add|change", SUBSYSTEM=="block", KERNEL!="loop*", KERNEL!="ram*", ENV{SNAP_AUTO_IMPORT}!="0" \
```

Comment:
I have the same question as Alfonso what will set SNAP_AUTO_IMPORT on the device?

Note that in general, we do not permit arbitrary udev rules to be installed on ubuntu core, there is a design effort underway at some point to allow certain kinds of udev rules to be installed via the gadget snap, but that is still in its infancy phase (this is on @pedronis plate, not sure how far along the design/thinking is on that). Due to this, I would be inclined to not introduce this functionality until we have a use case for it and we know how things will fit together.

Reply:
For instance if a user wants to use udisks2, they will provide it in a snap. The problem is that there will be error with udisks2 daemon and `snap auto-import` conflicting on mounting the filesystem. They can provide an udev rule with `UDISKS_AUTO=0` to disable auto mounting of some filesystems by udisks2. But there is nothing to disable `snap auto-import`. So that means there is no way to make some device being used only by udisks2. We need a way for them to disable `snap auto-import`. They can do it the way they want. For example they could do, only use `snap auto-import` if the fs label is "ASSERT":

```
# Only added/inserted block devices
ACTION=="add|change", SUBSYSTEM=="block", GOTO="start"
GOTO="end"
LABEL="start"
# Import blkid, or this is not useful if run after 60-persistent-storage.rules
IMPORT{builtin}="blkid"
# Check if the filesystem label is "ASSERT"
ENV{ID_FS_LABEL}=="ASSERT", GOTO="found_assert"
# If not, disable snap auto-import
ENV{SNAP_AUTO_IMPORT}=0
GOTO="end"
# If it is, disable udisks2 auto mount (can also use UDISKS_IGNORE)
LABEL="found_assert"
ENV{UDISKS_AUTO}="0"
LABEL="end"
```

Reply:
I guess what I'm trying to say is this step:

> They can provide an udev rule with UDISKS_AUTO=0 to disable auto mounting of some filesystems by udisks2

Is only doable if you have an unconfined shell on an Ubuntu Core device which isn't something we support, we really mainly focus on people modifying the system via confined snaps which only can install udev rules as per the interfaces that their snaps connect.